spark/src/test/scala/io/delta/sharing/spark/DeltaSharingSourceOffsetSuite.scala

```diff
@@ -0,0 +1,187 @@
+    var unknownVersion = 2
+
+    val json =
+      s"""
+         |{
+         |  "sourceVersion": $unknownVersion,
+         |  "tableVersion": 1,
+         |  "index": 1,
+         |  "isStartingVersion": true
+         |}
+      """.stripMargin
+    val e = intercept[IllegalStateException] {
+      DeltaSharingSourceOffset(UUID.randomUUID().toString, SerializedOffset(json))
+    }
+    assert(e.getMessage.contains("Please upgrade to a new release"))
+  }
+
+  test("DeltaSharingSourceOffset sourceVersion - invalid value") {
+    val json =
+      """
+        |{
+        |  "sourceVersion": "foo",
+        |  "tableVersion": 1,
+        |  "index": 1,
+        |  "isStartingVersion": true
+        |}
+      """.stripMargin
+    val e = intercept[IllegalStateException] {
+      DeltaSharingSourceOffset(UUID.randomUUID().toString, SerializedOffset(json))
+    }
+    for (msg <- Seq("foo", "invalid")) {
+      assert(e.getMessage.contains(msg))
+    }
+  }
+
+  test("DeltaSharingSourceOffset sourceVersion - missing ") {
+    val json =
+      """
+        |{
+        |  "tableVersion": 1,
+        |  "index": 1,
+        |  "isStartingVersion": true
+        |}
+      """.stripMargin
+    val e = intercept[IllegalStateException] {
+      DeltaSharingSourceOffset(UUID.randomUUID().toString, SerializedOffset(json))
+    }
+    for (msg <- Seq("Cannot find", "sourceVersion")) {
+      assert(e.getMessage.contains(msg))
+    }
+  }
+
+  test("DeltaSharingSourceOffset - unmatched reservoir id") {
+    val json =
+      s"""
+         |{
+         |  "tableId": "${UUID.randomUUID().toString}",
+         |  "sourceVersion": 1,
+         |  "tableVersion": 1,
+         |  "index": 1,
+         |  "isStartingVersion": true
+         |}
+      """.stripMargin
+    val e = intercept[IllegalStateException] {
+      DeltaSharingSourceOffset(UUID.randomUUID().toString, SerializedOffset(json))
+    }
+    for (msg <- Seq("delete", "checkpoint", "restart")) {
+      assert(e.getMessage.contains(msg))
+    }
+  }
+
+  test("DeltaSharingSourceOffset - validateOffsets") {
+    DeltaSharingSourceOffset.validateOffsets(
+      previousOffset = DeltaSharingSourceOffset(
+        sourceVersion = 1,
+        tableId = "foo",
+        tableVersion = 4,
+        index = 10,
+        isStartingVersion = false),
+      currentOffset = DeltaSharingSourceOffset(
+        sourceVersion = 1,
+        tableId = "foo",
+        tableVersion = 4,
+        index = 10,
+        isStartingVersion = false)
+    )
+    DeltaSharingSourceOffset.validateOffsets(
+      previousOffset = DeltaSharingSourceOffset(
+        sourceVersion = 1,
+        tableId = "foo",
+        tableVersion = 4,
+        index = 10,
+        isStartingVersion = false),
+      currentOffset = DeltaSharingSourceOffset(
+        sourceVersion = 1,
+        tableId = "foo",
+        tableVersion = 5,
+        index = 1,
+        isStartingVersion = false)
+    )
```

Comment:
consider adding the following:
- we only increase the index
- isStartingVersion is true for both (we can perhaps do a grid test for success where all combinations can be tested for isStartingVersion being false and true).

Reply:
updated.